packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -134,7 +134,13 @@ class TextEditingController extends ValueNotifier<TextEditingValue> {
   /// This constructor treats a null [value] argument as if it were
   /// [TextEditingValue.empty].
   TextEditingController.fromValue(TextEditingValue value)
-    : super(value ?? TextEditingValue.empty);
+    : assert(
+        value == null || !value.composing.isValid || value.isComposingRangeValid,
```

Comment:
It's somewhat confusing that `value.composing.isValid` and `value.isComposingRangeValid` seem so similar but mean different things.  I can't think of a way to make it more clear though 🤷

Reply:
I guess we could use null to represent an editing value that has no composing range. That would be a breaking change.

Reply:
Probably not worth it.